sdk/keyvault/Azure.Security.KeyVault.Shared/tests/KeyVaultTestEventListener.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.Tracing;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Security.KeyVault.Tests
+{
+    /// <summary>
+    /// Wraps an <see cref="AzureEventSourceListener"/> when running live tests.
+    /// </summary>
+    internal class KeyVaultTestEventListener : IDisposable
```

Comment:
Didn't we have a similar class already? cc @JoshLove-msft

Reply:
Perhaps, but it's a few lines using a KV-specific variable I already had to add to our tests.yml to account for track 1 test issues. The variable represents the intent, so I reused it and only want it running when set to 'Live' (in practice, no other value is supported anyway but figured I'd be specific to be safe).

Reply:
Why do we have a separate variable for KV test mode? Is the main one insufficient?

Reply:
Yes - I tried making this more generic for Search at https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/tests/Utilities/TestLogger.cs with the hope of moving it into the shared test code at some point.

Reply:
@pakrym as I explained in my comment: it was to solve an issue with track 1 tests. The existing variable couldn't be used. By making track 2 live, track 1 would run and doesn't work with our infrastructure. Making them work in our infrastructure is a much larger task and Alex didn't think it worth it.

@tg-msft thanks, but that's overkill. I just need to log request/responses to stdout so they get picked up in the TRX files that are already being generated and uploaded to the pipeline. I only want to those to run by default when that test env var is set. There's no need to copy this code from another service.

Reply:
The idea was not to copy but to make it shared and reuse.

Reply:
KeyVaultTestEventListener wouldn't work with track 1 so why not use the track 2 environment variable?

Reply:
It's a lot, but it's also so much nicer to use in practice.  It automatically dumps out logs for failing tests and doesn't flood the output with noise for the rest of the run.

Reply:
It's possible, but then also happens when people have their environments set for live runs. I'm not undefining the other variable for reasons stated, so right now it signifies "only running in our nightlies". You're the one on several previous occasions that wanted me to remove code that logged in more cases by default. You're asking me to put some back in now?

Reply:
But it doesn't do what I need.
1. I only want it looked conditionally for why I explained [previously](https://github.com/Azure/azure-sdk-for-net/pull/11170#discussion_r405837657).
2. I want output in successful and failed cases (the other test causing this isn't failing, but I can't identify it because I don't know what is deleting that particular secret).
3. It does more than I need, so I see no point in using it or taking the time to copy it to Azure.Core.Test and all the other authoring that would entail.

Reply:
I was asking you to remove inconsistent usages of logging but I'm all for setting up some common reusable logging thing.

Reply:
I just need this to be conditional. If anything, it uses more of your logging infrastructure. VSTest already attaches content written to stdout/stderr to the test in the TRX, so what I wrote was a simpler way of what @tg-msft did, and I need it for both successful and failed test cases. If anything, I recommend this pattern as opposed to a class that reformats message payload data. But I still only want this to log when running in our nightlies, so I'd still end up wrapping this anyway.